### PR TITLE
Adding new package vulkan-memory-allocator

### PR DESCRIPTION
**Pull request by felixaszx**
As the title said, VMA is widely use right now and also included inside the LunarG Vulkan SDK. So I think it is a good idea to include that in MSYS2 as well.

### CODE REVIEW

**Comment by ognevny** [comment]
the package uses cmake for build. why it isn't used here?

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Maintainer: Felixaszx (Felix Xing) felixxing3009@outlook.com
+
+_realname=VulkanMemoryAllocator
+pkgbase=mingw-w64-vulkan-memory-allocator
+pkgname=("${MINGW_PACKAGE_PREFIX}-vulkan-memory-allocator")
+pkgver=3.1.0
+pkgrel=1
+pkgdesc="The Vulkan® Memory Allocator (VMA) library provides a simple and easy to integrate API to help you allocate memory for Vulkan® buffer and image storage."
```
Comment (mmuetzel) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:8]: Potentially shorten the description to a one-liner and append "(mingw-w64)":
``` suggestion
pkgdesc="Memory allocation library for Vulkan API (mingw-w64)"
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url="https://gpuopen.com/vulkan-memory-allocator/"
+license=('MIT')
```
Comment (mmuetzel) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:12]:
``` suggestion
license=('spdx:MIT')
```

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+makedepends=('git')
+_commit='ff9ee68d7e31784c6fea3c864a235ad1f32ff026'
```
Comment (mmuetzel) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:14]: This value seems to be unused. Delete this line?

File: mingw-w64-vulkan-memory-allocator/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+pkgbase=mingw-w64-vulkan-memory-allocator
+pkgname=("${MINGW_PACKAGE_PREFIX}-vulkan-memory-allocator")
+pkgver=3.1.0
+pkgrel=1
+pkgdesc="The Vulkan® Memory Allocator (VMA) library provides a simple and easy to integrate API to help you allocate memory for Vulkan® buffer and image storage."
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url="https://gpuopen.com/vulkan-memory-allocator/"
+license=('MIT')
+makedepends=('git')
```
Comment (mmuetzel) [inline mingw-w64-vulkan-memory-allocator/PKGBUILD:13]: IIUC, you are downloading a tarball. Is `git` really a make dependency?

**Comment by felixaszx** [comment]
@mmuetzel I had remove unsed git dependency and kind of refine the descriptions.
The old one looks a bit long since it was copied from their website directly.

**Pull request merged**